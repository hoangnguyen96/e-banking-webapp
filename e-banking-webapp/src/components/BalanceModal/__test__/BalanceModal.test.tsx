import { render } from '@testing-library/react';

// Components
import { BalanceModal } from '..';

jest.mock('react-apexcharts', () => {
  return {
    __esModule: true,
    default: () => <div>Mock ApexCharts</div>,
  };
});

jest.mock('@/constants/rules', () => ({
  createStepSchema: jest.fn(),
}));

describe('BalanceModal component', () => {
  it('Should render snapshot correctly', () => {
    expect(
      render(
        <BalanceModal
          username='Pheroxios'
          isOpen={true}
          onClose={() => {}}
          currentBalance='90,000'
          totalInvestment='80,000'
        />,
      ),
    ).toMatchSnapshot();
  });
});
