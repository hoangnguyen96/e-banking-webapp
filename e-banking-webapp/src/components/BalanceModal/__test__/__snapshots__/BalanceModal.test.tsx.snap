// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BalanceModal component Should render snapshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-hidden="true"
    />
    <span
      aria-hidden="true"
      data-focus-scope-start="true"
      hidden=""
    />
    <div
      tabindex="-1"
    >
      <div
        aria-hidden="true"
        class="z-50 bg-overlay/50 backdrop-opacity-disabled w-screen h-screen fixed inset-0"
        style="opacity: 0;"
      />
      <div
        class="flex w-screen fixed inset-0 z-50 overflow-x-auto justify-center h-[--visual-viewport-height] items-start sm:items-start overflow-y-auto [--scale-enter:100%] [--scale-exit:100%] [--slide-enter:0px] [--slide-exit:80px] sm:[--scale-enter:100%] sm:[--scale-exit:103%] sm:[--slide-enter:0px] sm:[--slide-exit:0px]"
        data-slot="wrapper"
        style="--visual-viewport-height: 768px; opacity: 0; transform: translateY(var(--slide-exit)) scale(var(--scale-exit));"
      >
        <section
          aria-modal="true"
          class="flex flex-col relative z-50 w-full box-border bg-content1 outline-none mx-1 sm:mx-6 sm:my-16 max-w-4xl shadow-small my-16 !my-auto p-0 rounded-xl"
          data-dismissable="true"
          data-open="true"
          id=":r0:"
          role="dialog"
          tabindex="-1"
        >
          <div
            style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
          >
            <button
              aria-label="Dismiss"
              id="react-aria-:r5:"
              style="width: 1px; height: 1px;"
              tabindex="-1"
            />
          </div>
          <button
            aria-label="Close"
            class="z-0 group inline-flex items-center box-border whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] h-12 gap-3 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover border-transparent max-h-10 text-sm absolute appearance-none select-none top-1 right-1 rtl:left-1 rtl:right-[unset] p-2 rounded-full active:bg-default-200 tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 bg-transparent max-h-none min-w-[20px] w-[20x] text-foreground-200 justify-end hover:bg-transparent data-[focus=true]:bg-transparent"
            role="button"
            type="button"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 3H5C3.89543 3 3 3.89543 3 5V19C3 20.1046 3.89543 21 5 21H19C20.1046 21 21 20.1046 21 19V5C21 3.89543 20.1046 3 19 3Z"
                stroke="#264653"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M9 9L15 15"
                stroke="#264653"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M15 9L9 15"
                stroke="#264653"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
          <div
            class="absolute left-4 top-4"
          >
            <h4
              class="text-primary-200 text-base font-medium text-primary-200"
            >
              Good Evening,
               
              <span
                class="text-foreground-100 text-base font-normal"
              >
                Pheroxios
              </span>
            </h4>
          </div>
          <div
            class="flex rounded-xl bg-lightGraySolid"
          >
            <div
              class="flex flex-1 flex-col rounded-xl bg-white pb-[48px] pl-3 pr-[23px] pt-[56px]"
            >
              <div
                class="flex items-center justify-between"
              >
                <div
                  class="flex flex-col gap-2"
                >
                  <h5
                    class="text-foreground-100 text-sm font-semibold"
                  >
                    Current Balance
                  </h5>
                  <span
                    class="text-foreground-100 text-base font-extrabold"
                  >
                    $
                    90,000
                    <span
                      class="text-primary-200 text-xs font-semibold text-black/50"
                    >
                      /$
                    </span>
                  </span>
                </div>
                <div
                  class="flex flex-col gap-2"
                >
                  <h5
                    class="text-foreground-100 text-sm font-semibold"
                  >
                    Total Investments
                  </h5>
                  <span
                    class="text-foreground-100 text-base font-extrabold"
                  >
                    $
                    80,000
                  </span>
                </div>
              </div>
              <div
                class="relative mx-auto mt-9 flex h-[200px] w-[200px] items-center justify-center"
              >
                <div
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuenow="0"
                  aria-valuetext="0%"
                  class="flex flex-col justify-center gap-1 max-w-fit items-center"
                  id="react-aria-:r6:"
                  role="progressbar"
                >
                  <div
                    class="relative block"
                  >
                    <svg
                      class="z-0 relative overflow-hidden text-primary w-[200px] h-[200px] drop-shadow-md"
                      fill="none"
                      stroke-width="3"
                      viewBox="0 0 32 32"
                    >
                      <circle
                        class="h-full stroke-[#2A9D8F]/10"
                        cx="16"
                        cy="16"
                        r="13"
                        role="presentation"
                        stroke-dasharray="81.68140899333463 81.68140899333463"
                        stroke-dashoffset="0"
                        stroke-linecap="round"
                        transform="rotate(-90 16 16)"
                      />
                      <circle
                        class="h-full transition-all !duration-500 stroke-[#2A9D8F]"
                        cx="16"
                        cy="16"
                        r="13"
                        role="presentation"
                        stroke-dasharray="81.68140899333463 81.68140899333463"
                        stroke-dashoffset="81.68140899333463"
                        stroke-linecap="round"
                        transform="rotate(-90 16 16)"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="absolute flex flex-col items-center"
                >
                  <p
                    class="text-foreground-100 text-center text-sm font-medium"
                  >
                    Spend
                  </p>
                  <span
                    class="text-foreground-100 text-base text-center font-extrabold"
                  >
                    0
                    %
                  </span>
                </div>
              </div>
            </div>
            <div
              class="flex flex-1 flex-col pb-[28px] pl-4 pr-3 pt-[56px]"
            >
              <h4
                class="text-foreground-100 text-base font-medium"
              >
                My Cards
              </h4>
              <div
                class="my-4"
              >
                <div
                  class="flex gap-[9px]"
                >
                  <div
                    class="hover:cursor-pointer"
                  >
                    <div
                      class="relative overflow-hidden text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none flex h-[88px] w-full min-w-[111px] flex-col justify-between rounded-[6px] border-0 p-2 bg-secondary-300"
                      tabindex="-1"
                    >
                      <div
                        class="flex gap-3 p-0"
                      >
                        <div
                          class="flex flex-1 items-center justify-between"
                        >
                          <p
                            class="text-foreground-200 font-medium text-xs"
                          >
                            Main
                          </p>
                          <svg
                            class="w-[25px] h-[15px]"
                            fill="none"
                            height="35"
                            viewBox="0 0 59 35"
                            width="59"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              clip-rule="evenodd"
                              d="M41.0762 34.2205C50.791 34.2205 58.6663 26.723 58.6663 17.4744C58.6663 8.22576 50.791 0.728271 41.0762 0.728271C36.7224 0.728271 32.738 2.23419 29.6664 4.72876C26.5948 2.23426 22.6104 0.72838 18.2566 0.72838C8.54186 0.72838 0.666504 8.22586 0.666504 17.4745C0.666504 26.7231 8.54186 34.2206 18.2566 34.2206C22.6105 34.2206 26.5949 32.7147 29.6665 30.2201C32.7381 32.7146 36.7224 34.2205 41.0762 34.2205Z"
                              fill="#ED0006"
                              fill-rule="evenodd"
                            />
                            <path
                              clip-rule="evenodd"
                              d="M29.6665 30.2197C33.4484 27.1482 35.8465 22.5778 35.8465 17.4742C35.8465 12.3707 33.4484 7.80033 29.6665 4.72879C32.7381 2.23413 36.7226 0.728149 41.0766 0.728149C50.7913 0.728149 58.6667 8.22563 58.6667 17.4742C58.6667 26.7229 50.7913 34.2203 41.0766 34.2203C36.7226 34.2203 32.7381 32.7144 29.6665 30.2197Z"
                              fill="#F9A000"
                              fill-rule="evenodd"
                            />
                            <path
                              clip-rule="evenodd"
                              d="M29.6668 30.2192C33.4488 27.1476 35.8469 22.5773 35.8469 17.4737C35.8469 12.3702 33.4488 7.79981 29.6668 4.72827C25.8849 7.79981 23.4868 12.3702 23.4868 17.4737C23.4868 22.5773 25.8849 27.1476 29.6668 30.2192Z"
                              fill="#FF5E00"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        class="flex items-center justify-between"
                      >
                        <p
                          class="text-foreground-200 font-medium text-2xs opacity-100"
                        >
                          .... 
                          6789
                        </p>
                        <p
                          class="text-foreground-200 font-medium text-2xs opacity-100"
                        >
                          06/24
                        </p>
                      </div>
                    </div>
                  </div>
                  <div
                    class="hover:cursor-pointer"
                  >
                    <div
                      class="relative overflow-hidden text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none flex h-[88px] w-full min-w-[111px] flex-col justify-between rounded-[6px] border-0 p-2 bg-background-500"
                      tabindex="-1"
                    >
                      <div
                        class="flex gap-3 p-0"
                      >
                        <div
                          class="flex flex-1 items-center justify-between"
                        >
                          <p
                            class="text-foreground-100 font-medium text-xs"
                          >
                            Savings
                          </p>
                          <svg
                            class="w-[25px] h-[15px]"
                            fill="none"
                            height="35"
                            viewBox="0 0 59 35"
                            width="59"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              clip-rule="evenodd"
                              d="M41.0762 34.2205C50.791 34.2205 58.6663 26.723 58.6663 17.4744C58.6663 8.22576 50.791 0.728271 41.0762 0.728271C36.7224 0.728271 32.738 2.23419 29.6664 4.72876C26.5948 2.23426 22.6104 0.72838 18.2566 0.72838C8.54186 0.72838 0.666504 8.22586 0.666504 17.4745C0.666504 26.7231 8.54186 34.2206 18.2566 34.2206C22.6105 34.2206 26.5949 32.7147 29.6665 30.2201C32.7381 32.7146 36.7224 34.2205 41.0762 34.2205Z"
                              fill="#ED0006"
                              fill-rule="evenodd"
                            />
                            <path
                              clip-rule="evenodd"
                              d="M29.6665 30.2197C33.4484 27.1482 35.8465 22.5778 35.8465 17.4742C35.8465 12.3707 33.4484 7.80033 29.6665 4.72879C32.7381 2.23413 36.7226 0.728149 41.0766 0.728149C50.7913 0.728149 58.6667 8.22563 58.6667 17.4742C58.6667 26.7229 50.7913 34.2203 41.0766 34.2203C36.7226 34.2203 32.7381 32.7144 29.6665 30.2197Z"
                              fill="#F9A000"
                              fill-rule="evenodd"
                            />
                            <path
                              clip-rule="evenodd"
                              d="M29.6668 30.2192C33.4488 27.1476 35.8469 22.5773 35.8469 17.4737C35.8469 12.3702 33.4488 7.79981 29.6668 4.72827C25.8849 7.79981 23.4868 12.3702 23.4868 17.4737C23.4868 22.5773 25.8849 27.1476 29.6668 30.2192Z"
                              fill="#FF5E00"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        class="flex items-center justify-between"
                      >
                        <p
                          class="text-foreground-100 font-medium text-2xs opacity-50"
                        >
                          .... 
                          6789
                        </p>
                        <p
                          class="text-foreground-100 font-medium text-2xs opacity-50"
                        >
                          06/24
                        </p>
                      </div>
                    </div>
                  </div>
                  <div
                    class="hover:cursor-pointer"
                  >
                    <div
                      class="relative overflow-hidden text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none flex h-[88px] w-full min-w-[111px] flex-col justify-between rounded-[6px] border-0 p-2 bg-background-500"
                      tabindex="-1"
                    >
                      <div
                        class="flex gap-3 p-0"
                      >
                        <div
                          class="flex flex-1 items-center justify-between"
                        >
                          <p
                            class="text-foreground-100 font-medium text-xs"
                          >
                            Checkings
                          </p>
                          <svg
                            class="w-[25px] h-[15px]"
                            fill="none"
                            height="35"
                            viewBox="0 0 59 35"
                            width="59"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              clip-rule="evenodd"
                              d="M41.0762 34.2205C50.791 34.2205 58.6663 26.723 58.6663 17.4744C58.6663 8.22576 50.791 0.728271 41.0762 0.728271C36.7224 0.728271 32.738 2.23419 29.6664 4.72876C26.5948 2.23426 22.6104 0.72838 18.2566 0.72838C8.54186 0.72838 0.666504 8.22586 0.666504 17.4745C0.666504 26.7231 8.54186 34.2206 18.2566 34.2206C22.6105 34.2206 26.5949 32.7147 29.6665 30.2201C32.7381 32.7146 36.7224 34.2205 41.0762 34.2205Z"
                              fill="#ED0006"
                              fill-rule="evenodd"
                            />
                            <path
                              clip-rule="evenodd"
                              d="M29.6665 30.2197C33.4484 27.1482 35.8465 22.5778 35.8465 17.4742C35.8465 12.3707 33.4484 7.80033 29.6665 4.72879C32.7381 2.23413 36.7226 0.728149 41.0766 0.728149C50.7913 0.728149 58.6667 8.22563 58.6667 17.4742C58.6667 26.7229 50.7913 34.2203 41.0766 34.2203C36.7226 34.2203 32.7381 32.7144 29.6665 30.2197Z"
                              fill="#F9A000"
                              fill-rule="evenodd"
                            />
                            <path
                              clip-rule="evenodd"
                              d="M29.6668 30.2192C33.4488 27.1476 35.8469 22.5773 35.8469 17.4737C35.8469 12.3702 33.4488 7.79981 29.6668 4.72827C25.8849 7.79981 23.4868 12.3702 23.4868 17.4737C23.4868 22.5773 25.8849 27.1476 29.6668 30.2192Z"
                              fill="#FF5E00"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        class="flex items-center justify-between"
                      >
                        <p
                          class="text-foreground-100 font-medium text-2xs opacity-50"
                        >
                          .... 
                          6789
                        </p>
                        <p
                          class="text-foreground-100 font-medium text-2xs opacity-50"
                        >
                          06/24
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <p
                class="text-foreground-100 text-xs max-w-[180px] font-normal"
              >
                You have a total of
                 
                <span
                  class="text-foreground-100 text-xs font-bold"
                >
                  $
                </span>
                 
                available in your wallet
              </p>
              <div
                class="mt-2.5"
              >
                <div
                  class="flex flex-col relative overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none max-w-[250px] bg-background-300 p-[10px]"
                  tabindex="-1"
                >
                  <div
                    class="flex z-10 w-full justify-start items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large p-0"
                  >
                    <p
                      class="text-foreground-200 text-base font-medium"
                    >
                      Expense Analysis
                    </p>
                  </div>
                  <div
                    class="mt-2 flex gap-4"
                  >
                    <div
                      class="grid grid-cols-2 content-center gap-x-5 gap-y-3"
                    >
                      <div
                        class="flex h-3 items-center gap-[5px]"
                      >
                        <div
                          class="h-[8px] w-[8px] rounded-full bg-[#8AB17D]"
                        />
                        <p
                          class="text-primary-200 font-medium text-2xs text-white opacity-50"
                        >
                          Amazon
                        </p>
                      </div>
                      <div
                        class="flex h-3 items-center gap-[5px]"
                      >
                        <div
                          class="h-[8px] w-[8px] rounded-full bg-[#2A9D8F]"
                        />
                        <p
                          class="text-primary-200 font-medium text-2xs text-white opacity-50"
                        >
                          Netflix
                        </p>
                      </div>
                      <div
                        class="flex h-3 items-center gap-[5px]"
                      >
                        <div
                          class="h-[8px] w-[8px] rounded-full bg-[#F4A261]"
                        />
                        <p
                          class="text-primary-200 font-medium text-2xs text-white opacity-50"
                        >
                          Apple
                        </p>
                      </div>
                      <div
                        class="flex h-3 items-center gap-[5px]"
                      >
                        <div
                          class="h-[8px] w-[8px] rounded-full bg-[#E9C46A]"
                        />
                        <p
                          class="text-primary-200 font-medium text-2xs text-white opacity-50"
                        >
                          Figma
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
          >
            <button
              aria-label="Dismiss"
              id="react-aria-:r9:"
              style="width: 1px; height: 1px;"
              tabindex="-1"
            />
          </div>
        </section>
      </div>
    </div>
    <span
      aria-hidden="true"
      data-focus-scope-end="true"
      hidden=""
    />
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
